examples/Basics/user_callable.py

```diff
@@ -0,0 +1,176 @@
+## Example for a user defined callable
+
+# This example is an extenssion of the recommenders example.
+# It implements an End-to-End example, that showes how to use the user defined callable in pure recommenders.
+
+# This examples assumes some basic familiarity with using BayBE.
+# We refer to [`campaign`](./campaign.md) for a more general and basic example.
+
+### Necessary imports for this example
+
+import pickle
+from typing import Optional
+
+from pandas import DataFrame
+
+from baybe import Campaign
+from baybe.objectives import SingleTargetObjective
+from baybe.parameters import NumericalDiscreteParameter, SubstanceParameter
+from baybe.recommenders import (
+    RandomRecommender,
+    SequentialGreedyRecommender,
+    TwoPhaseMetaRecommender,
+)
+from baybe.searchspace import SearchSpace
+from baybe.surrogates import GaussianProcessSurrogate
+from baybe.targets import NumericalTarget
+from baybe.utils.dataframe import add_fake_results
+
+# Per default the initial recommender chosen is a random recommender.
+INITIAL_RECOMMENDER = RandomRecommender()
+
+### Surrogate models
+
+# This model uses available data to model the objective function as well as the uncertainty.
+# The surrogate model is then used by the acquisition function to make recommendations.
+
+# Per default a Gaussian Process is used
+SURROGATE_MODEL = GaussianProcessSurrogate()
+
+### Acquisition function
+
+# This function looks for points where measurements of the target value could improve the model.
+
+# Note that the qvailability of the acquisition functions might depend on the `batch_size`:
+#   - If `batch_size` is set to 1, all available acquisition functions can be chosen
+#   - If a larger value is chosen, only those that allow batching.
+#       That is, 'q'-variants of the acquisition functions must be chosen.
+
+# The default acquisition function is q-Expected Improvement.
+
+ACQ_FUNCTION = "qEI"
+
+
+### Callable function
+def callable_test(
```

Comment:
This function does two things: Printing all of the internal data and pickling the recommender object. Ideally, this function should only do one thing, so either pickling something or printing stuff. For example, a function just printing the training data would already be enough imo: Since all of the other parameters are already available, I think it is not strictly necessary to also print them here.

Reply:
But if I only print the training data the other parameters will not be used. Do they still need to be part of the signature of `callable_test` or not?

Reply:
I think we discussed this in the meeting, if something is unclear just ask.